tests/test_table_name.py

```diff
@@ -4,6 +4,54 @@
 from sqlalchemy.exc import ArgumentError
 from sqlalchemy.ext.declarative import declared_attr
 
+from flask_sqlalchemy.model import camel_to_snake_case
+
+
+def test_camel_to_snake_case():
+    input = [
+        "ABC",
+        "PreABC",
+        "ABCPost",
+        "PreABCPost",
+        "VP9Codec",
+        "HTTP2RequestSession",
+        "UserST4",
+        "HTTP2ClientType3EncoderParametersSSE",
+        "LONGName4TestingCamelCase2snake_caseKEK",
+        "FooBarSSE2",
+        "AlarmMessageSS2SignalTransformer",
+        "AstV2Node",
+        "HTTPResponseCodeXYZ",
+        "getHTTPresponseCode",
+        "__test__Method",
+        "get2HTTPResponse123Code",
+        "MultinomialNB",
+        "Fun?!awesome",
+    ]
+    output = [
+        "abc",
+        "pre_abc",
+        "abc_post",
+        "pre_abc_post",
+        "vp9_codec",
+        "http2_request_session",
+        "user_st4",
+        "http2_client_type3_encoder_parameters_sse",
+        "long_name4_testing_camel_case2snake_case_kek",
+        "foo_bar_sse2",
+        "alarm_message_ss2_signal_transformer",
+        "ast_v2_node",
+        "http_response_code_xyz",
+        "get_htt_presponse_code",
```

Comment:
YUCK ;) but I guess we can't really avoid such edge cases

Reply:
To be fair, the original name was yuck too.